### PR TITLE
Grafana installation fixed

### LINKED ISSUE #914: Grafana DB migration during install
RPi4 2G, own built image, latest openHABian.
Got this error (never seen before):
```
Resetting Grafana admin password...
$ grafana-cli admin reset-admin-password admin
INFO[05-25|22:27:01] Connecting to DB                         logger=sqlstore dbtype=sqlite3
INFO[05-25|22:27:01] Starting DB migration                    logger=migrator
INFO[05-25|22:27:01] Executing migration                      logger=migrator id="Rename table data_source to data_source_v1 - v1"
INFO[05-25|22:27:02] Executing migration                      logger=migrator id="create data_source table v2"
INFO[05-25|22:27:02] Executing migration                      logger=migrator id="create index IDX_data_source_org_id - v2"
EROR[05-25|22:27:02] Executing migration failed               logger=migrator id="create index IDX_data_source_org_id - v2" error="index IDX_data_source_org_id already exists"
EROR[05-25|22:27:02] Exec failed                              logger=migrator error="index IDX_data_source_org_id already exists" sql="CREATE INDEX `IDX_data_source_org_id` ON `data_source` (`org_id`);"
Error: ✗ failed to initialize SQL engine: Migration failed err: Failed to roll back transaction due to error: %!s(<nil>): database is locked
```

Maybe just a coincidence, needs to be investagated.

### PR DESCRIPTION
**Pull request by holgerfriedrich**
* password reset only if grafana.db exists to avoid side effects,
  fixes #914
* adapted bats test to check connection to influxdb
* adapted bats test to store settings in grafana, validating connection
  and settings file grafana.db

Signed-off-by: Holger Friedrich <mail@holger-friedrich.de>

### CODE REVIEW

**Comment by holgerfriedrich** [comment]
@mstormi any idea how we could make the output of the bats tests visible in CI?
https://travis-ci.org/github/holgerfriedrich/openhabian/jobs/691133185
This is all I see from the destructive tests:

`$ docker run --name "bats-dest-tests" -d openhabian/bats-openhabian bash -c '/usr/local/bin/bats -r -f "destructive-." .'
dfe3124d8215b6c9bb6da1698d867108dddc7f7251ec1b6a88b508e43e7e5746
The command "docker run --name "bats-dest-tests" -d openhabian/bats-openhabian bash -c '/usr/local/bin/bats -r -f "destructive-." .'" exited with 0.`

Compare to output when I start it locally:
`$ sudo bats  functions/influxdb+grafana.bats
    InfluxDB installation starting...                                       1/2
   InfluxDB installation successful.
   InfluxDB service running.
   InfluxDB service is responding.
 ✓ destructive-influxDB_install
    Grafana installation starting...                                        2/2
   Grafana installation successful.
   Grafana service running.
   Grafana settings sucessfully changed.
   Grafana re-installation successful.
   Grafana settings sucessfully changed.
 ✓ destructive-grafana_install
2 tests, 0 failures
`

**Comment by mstormi** [comment]
> @mstormi any idea how we could make the output of the bats tests visible in CI?

Not really. You made me aware that the output is missing in all builds. I opened #917.
Note it actually works for the dev- tests, see .1 build logs.
Comparing dev-* and destructive-*, the only major difference I noticed is the docker parameters in .travis.yml.

**Comment by mstormi** [comment]
Ok, I just merged #918 now you should see your output.

Then again, a whole bunch of new problems to solve (BATS tests to fail on java and other)

File: functions/influxdb+grafana.bash [hunk 1]
```diff
@@ -286,11 +286,18 @@ influxdb_install() {
...
   echo -n "Installing Grafana... "
   cond_redirect wget -O - https://packages.grafana.com/gpg.key | apt-key add - || FAILED=2
   echo "deb https://packages.grafana.com/oss/deb stable main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
   cond_redirect apt-get update || FAILED=1
   cond_redirect apt-get -y install grafana || FAILED=2
+  if [ -e /var/lib/grafana/grafana.db ]; then
+    echo -n "Detected existing Grafana install..."
+  else
+    # shellcheck disable=SC2034
+    grafana_fist_time=true
```
Comment (mstormi) [inline functions/influxdb+grafana.bash:299]: "fist"
-> does the code work with that or is the check inverted, too ??
shellcheck hint probably not needed as well

File: functions/influxdb+grafana.bash [hunk 2]
```diff
@@ -299,10 +306,15 @@ grafana_install(){
...
   cond_echo ""
 
-  # password reset required if Grafana password was already set before (no first-time install)
-  echo -n "Resetting Grafana admin password... "
-  cond_redirect grafana-cli admin reset-admin-password admin || FAILED=2
-  if [ $FAILED -eq 2 ]; then echo -n "FAILED "; return 2; else echo -n "OK "; fi
+  # password reset required if Grafana password was already set before (no first-time install);
+  # do not set password for fresh installs because grafana.db will be created with root permissions
+  # which causes followup issues
+  if [ -z "$grafana_first_time" ]; then
```
Comment (mstormi) [inline functions/influxdb+grafana.bash:312]: I actually just fixed #702 (which was about removing the FAILED variable)

Please rework InfluxDB/Grafana install to be consistent and fail right away, too (and remember to return, not exit !)

File: functions/influxdb+grafana.bash (hunk 1 above)
Comment (holgerfriedrich) [reply functions/influxdb+grafana.bash]: too late for coding :-/ shellcheck is always right, I should have known
done.

Comment (mstormi) [reply functions/influxdb+grafana.bash]: > too late for coding :-/

Move on to some reviewing then ;-)
#861, #830, #807, and most important #882 
Juggling with too many open PRs can easily end up in chaos

File: functions/influxdb+grafana.bash (hunk 2 above)
Comment (holgerfriedrich) [reply functions/influxdb+grafana.bash:311]: I would not mix this up right now and better get this merged without further delay, as we have a broken installer right now.
My proposal: I will try the latest stuff on my test installation (new install, re-install). Then see if the bats tests work locally and then we could merge. Fixing the CI is in #917.

Comment (mstormi) [reply functions/influxdb+grafana.bash:311]: ok go ahead. Let me know when you want me to merge

**Comment by holgerfriedrich** [comment]
@mstormi ready to merge, local bats test is fine, manual test as well. CI still failing.
Grafana seems to be a bit tricky when using it during the first startup, added waiting for http and additionally a sleep.

File: functions/influxdb+grafana.bash [hunk 3]
```diff
@@ -290,23 +290,31 @@ grafana_install(){
   cond_redirect wget -O - https://packages.grafana.com/gpg.key | apt-key add - || FAILED=2
   echo "deb https://packages.grafana.com/oss/deb stable main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
   cond_redirect apt-get update || FAILED=1
-  cond_redirect apt-get -y install grafana || FAILED=2
+  cond_redirect apt-get -y install grafana || FAILED=
```
Comment (mstormi) [inline functions/influxdb+grafana.bash:293]: FAILED=(empty) intentional ?

Comment (holgerfriedrich) [reply functions/influxdb+grafana.bash]: thanks, no this was not intended

**Pull request merged**